source/served/extension.d

```diff
@@ -111,6 +112,11 @@ void changedConfig(ConfigWorkspace target, string[] paths, served.types.Configur
 				backend.get!DCDComponent(workspaceFs)
 					.addImports(config.d.projectImportPaths.map!(a => a.userPath).array);
 			break;
+		case "d.ccdbPath":
+			if (backend.has!ClangCompilationDatabaseComponent(workspaceFs))
+				backend.get!ClangCompilationDatabaseComponent(workspaceFs)
+					.setPath(config.d.ccdbPath.userPath);
```

Comment:
this change event will only trigger if the config changes from the default - so it would probably be a good idea to also include the config value in the CCDB component config.

Additionally the config change event may trigger before the instance is loaded, so the setPath function would never be run there with the current implementation.